### ISSUE TITLE
rename file name no update list

### ISSUE THREAD

**Issue by willpyshan13** [nextcloud/android#7299]
Steps to reproduce
open app
rename file from all file
list is not updating

Expected behaviour

list should update

Actual behaviour

list is not updating

**Comment by aqaqsubin** [comment]
I also see this issue.
Device Model : Samsung Galaxy s10 e
Android version 10
Nextcloud app 3.13.1
Logs 
```
2020-11-12T23:11:34.636+0900;D;LogsActivity;onPause() ending
2020-11-12T23:11:34.647+0900;D;SettingsActivity;onStart() starting
2020-11-12T23:11:34.653+0900;D;SettingsActivity;onResume() starting
2020-11-12T23:11:35.004+0900;D;LogsActivity;onStop() ending
2020-11-12T23:11:35.005+0900;D;LogsActivity;onDestroy() ending
2020-11-12T23:11:35.386+0900;D;SettingsActivity;onPause() ending
2020-11-12T23:11:35.396+0900;V;BaseActivity;onRestart() start
2020-11-12T23:11:35.398+0900;D;FileDisplayActivity;onStart() starting
2020-11-12T23:11:35.410+0900;V;ArbitraryDataProvider;Updating arbitrary data with cloud id: global key: APP_OPENED_COUNT value: 2
2020-11-12T23:11:35.418+0900;D;OwnCloudClient #0;REQUEST GET /ocs/v1.php/cloud/user
2020-11-12T23:11:35.422+0900;D;AdvancedSslSocketFactory;Creating SSL Socket with remote efss.qloud.my:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@60e6bf2
2020-11-12T23:11:35.423+0900;D;AdvancedSslSocketFactory; ... with connection timeout 60000 and socket timeout 60000
2020-11-12T23:11:35.424+0900;D;cache_test_DISK_;image read from disk -1365829606
2020-11-12T23:11:35.425+0900;I;ServerNameIndicator;SSLSocket implementation: org.conscrypt.Java8FileDescriptorSocket
2020-11-12T23:11:35.425+0900;I;ServerNameIndicator;SNI done, hostname: efss.qloud.my
2020-11-12T23:11:35.433+0900;V;FileDisplayActivity;onResume() start
2020-11-12T23:11:35.433+0900;D;cache_test_DISK_;image read from disk -1365829606
2020-11-12T23:11:35.434+0900;D;FileDisplayActivity;onResume() starting
2020-11-12T23:11:35.472+0900;V;FileDisplayActivity;onResume() end
2020-11-12T23:11:35.479+0900;E;ThemeUtils;Error setting TextView handles color
2020-11-12T23:11:35.479+0900;E;ThemeUtils;Error setting TextView handles color
2020-11-12T23:11:35.485+0900;V;OCFileListFragment;Kept the options menu default structure
2020-11-12T23:11:35.508+0900;D;cache_test_DISK_;image read from disk 1315601912
2020-11-12T23:11:35.521+0900;D;cache_test_DISK_;image read from disk -1510636405
2020-11-12T23:11:35.523+0900;D;cache_test_DISK_;image read from disk -1503226283
2020-11-12T23:11:35.874+0900;D;SettingsActivity;onStop() ending
2020-11-12T23:11:35.875+0900;D;SettingsActivity;onDestroy() ending
2020-11-12T23:11:35.997+0900;D;OwnCloudClient #0;REQUEST GET /index.php/apps/external/icons/settings.svg
2020-11-12T23:11:36.001+0900;D;OwnCloudClient #0;REQUEST GET /status.php
2020-11-12T23:11:36.002+0900;D;AdvancedSslSocketFactory;Creating SSL Socket with remote efss.qloud.my:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@d11f4cc
2020-11-12T23:11:36.002+0900;D;AdvancedSslSocketFactory; ... with connection timeout 60000 and socket timeout 60000
2020-11-12T23:11:36.003+0900;I;ServerNameIndicator;SNI done, hostname: efss.qloud.my
2020-11-12T23:11:36.191+0900;I;com.owncloud.android.utils.glide.HttpStreamFetcher;Cleanup
2020-11-12T23:11:36.932+0900;D;UpdateOCVersionOperation;Got new OC version 19.0.4.2
2020-11-12T23:11:36.933+0900;I;UpdateOCVersionOperation;Check for update of Nextcloud server version at https://efss.qloud.my/remote.php/webdav: Operation finished with HTTP status code -1 (success)
2020-11-12T23:11:36.942+0900;D;OwnCloudClient #0;REQUEST GET /ocs/v2.php/cloud/capabilities
2020-11-12T23:11:37.359+0900;D;GetCapabilitiesRemoteOperation;Capabilities not modified
2020-11-12T23:11:37.360+0900;D;GetCapabilitiesRemoteOperation;*** Get Capabilities completed 
2020-11-12T23:11:37.389+0900;D;OwnCloudClient #0;REQUEST GET /ocs/v1.php/cloud/user
2020-11-12T23:11:37.678+0900;I;RefreshFolderOperation;Got display name: com.owncloud.android.lib.common.UserInfo@4242419
2020-11-12T23:11:37.678+0900;D;RefreshFolderOperation;Checking changes in aqaq925120k@gmail.com@efss.qloud.my/
2020-11-12T23:11:37.683+0900;D;OwnCloudClient #0;REQUEST PROPFIND /remote.php/webdav/
2020-11-12T23:11:38.166+0900;I;RefreshFolderOperation;Checked aqaq925120k@gmail.com@efss.qloud.my/ : changed
2020-11-12T23:11:38.170+0900;D;OwnCloudClient #0;REQUEST PROPFIND /remote.php/webdav/
2020-11-12T23:11:38.769+0900;I;ReadFolderRemoteOperation;Synchronized /: Operation finished with HTTP status code 207 (success)
2020-11-12T23:11:38.769+0900;D;RefreshFolderOperation;Synchronizing aqaq925120k@gmail.com@efss.qloud.my/
2020-11-12T23:11:38.774+0900;D;RefreshFolderOperation;Remote folder / changed - starting update of local data 
2020-11-12T23:11:38.786+0900;D;FileDataStorageManager;Saving folder / with 13 children and 0 files to remove
2020-11-12T23:11:38.834+0900;D;FileDataStorageManager;Sending 14 operations to FileContentProvider
2020-11-12T23:11:38.835+0900;D;FileContentProvider;applying batch in provider com.owncloud.android.providers.FileContentProvider@909584a (temporary: false)
2020-11-12T23:11:38.852+0900;D;FileContentProvider;applied batch in provider com.owncloud.android.providers.FileContentProvider@909584a
2020-11-12T23:11:38.858+0900;D;RefreshFolderOperation;Send broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_CONTENTS_SYNCED
2020-11-12T23:11:38.860+0900;D;OwnCloudClient #0;REQUEST GET /ocs/v2.php/apps/files_sharing/api/v1/shares
2020-11-12T23:11:38.860+0900;D;FileDisplayActivity;Received broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_CONTENTS_SYNCED
2020-11-12T23:11:38.906+0900;D;FileDisplayActivity;Setting progress visibility to true
2020-11-12T23:11:38.925+0900;D;cache_test_DISK_;image read from disk 1315601912
2020-11-12T23:11:38.939+0900;D;cache_test_DISK_;image read from disk -1510636405
2020-11-12T23:11:38.940+0900;D;cache_test_DISK_;image read from disk -1503226283
2020-11-12T23:11:39.267+0900;D;GetSharesForFileRemoteOperation;Got 0 shares
2020-11-12T23:11:39.287+0900;D;FileDataStorageManager;Sending 13 operations to FileContentProvider
2020-11-12T23:11:39.288+0900;D;FileContentProvider;applying batch in provider com.owncloud.android.providers.FileContentProvider@909584a (temporary: false)
2020-11-12T23:11:39.307+0900;D;FileContentProvider;applied batch in provider com.owncloud.android.providers.FileContentProvider@909584a
2020-11-12T23:11:39.307+0900;D;RefreshFolderOperation;Send broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_SHARES_SYNCED
2020-11-12T23:11:39.311+0900;D;FileDisplayActivity;Received broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_SHARES_SYNCED
2020-11-12T23:11:39.366+0900;D;FileDisplayActivity;Setting progress visibility to false
2020-11-12T23:11:39.392+0900;D;cache_test_DISK_;image read from disk 1315601912
2020-11-12T23:11:39.405+0900;D;cache_test_DISK_;image read from disk -1510636405
2020-11-12T23:11:39.407+0900;D;cache_test_DISK_;image read from disk -1503226283
2020-11-12T23:11:40.733+0900;E;ThemeUtils;Error setting TextView handles color
2020-11-12T23:11:45.067+0900;D;FileActivity;show loading dialog
2020-11-12T23:11:45.112+0900;D;OperationsService;Starting command with id 2
2020-11-12T23:11:45.124+0900;E;RenameFileOperation;Unable to create parent folder /storage/emulated/0/Android/media/com.nextcloud.client/nextcloud/tmp
2020-11-12T23:11:45.129+0900;D;OwnCloudClient #0;REQUEST HEAD /remote.php/webdav/Audio5.pcm
2020-11-12T23:11:45.131+0900;D;AdvancedSslSocketFactory;Creating SSL Socket with remote efss.qloud.my:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@d11f4cc
2020-11-12T23:11:45.131+0900;D;AdvancedSslSocketFactory; ... with connection timeout 60000 and socket timeout 60000
2020-11-12T23:11:45.136+0900;I;ServerNameIndicator;SSLSocket implementation: org.conscrypt.Java8FileDescriptorSocket
2020-11-12T23:11:45.136+0900;I;ServerNameIndicator;SNI done, hostname: efss.qloud.my
2020-11-12T23:11:45.282+0900;D;cache_test_DISK_;image read from disk -1510636405
2020-11-12T23:11:45.687+0900;D;OwnCloudClient;HEAD to /Audio5.pcm finished with HTTP status 404(FAIL)
2020-11-12T23:11:45.687+0900;D;OwnCloudClient #0;REQUEST MOVE /remote.php/webdav/5.pcm
2020-11-12T23:11:46.333+0900;I;RenameFileRemoteOperation;Rename /5.pcm to /Audio5.pcm: Operation finished with HTTP status code 201 (success)
2020-11-12T23:11:46.333+0900;D;OCFile;OCFile name changing from /5.pcm
2020-11-12T23:11:46.333+0900;D;OCFile;OCFile name changed to /Audio5.pcm
2020-11-12T23:11:46.345+0900;D;FileActivity;Received result of operation in FileActivity - common behaviour for all the FileActivities 
2020-11-12T23:11:46.345+0900;D;FileActivity;dismiss loading dialog
2020-11-12T23:11:46.347+0900;D;OperationsService;Called 1 listeners
2020-11-12T23:11:46.347+0900;D;OperationsService;Stopping after command with id 2
2020-11-12T23:11:46.430+0900;D;cache_test_DISK_;image read from disk 1315601912
2020-11-12T23:11:46.443+0900;D;cache_test_DISK_;image read from disk -1510636405
2020-11-12T23:11:46.445+0900;D;cache_test_DISK_;image read from disk -1503226283
2020-11-12T23:11:49.546+0900;D;OwnCloudClient #0;REQUEST GET /index.php/apps/external/icons/settings.svg
2020-11-12T23:11:49.744+0900;I;com.owncloud.android.utils.glide.HttpStreamFetcher;Cleanup
2020-11-12T23:11:49.982+0900;E;ThemeUtils;Error setting TextView handles color
2020-11-12T23:11:49.983+0900;E;ThemeUtils;Error setting TextView handles color
2020-11-12T23:11:49.996+0900;V;OCFileListFragment;Kept the options menu default structure
2020-11-12T23:11:54.457+0900;E;ThemeUtils;Error setting TextView handles color
2020-11-12T23:11:54.457+0900;E;ThemeUtils;Error setting TextView handles color
2020-11-12T23:11:54.463+0900;V;OCFileListFragment;Kept the options menu default structure
2020-11-12T23:11:54.473+0900;V;FileDisplayActivity;onPause() start
2020-11-12T23:11:54.474+0900;D;FileDisplayActivity;onPause() ending
2020-11-12T23:11:54.475+0900;V;FileDisplayActivity;onPause() end
2020-11-12T23:11:54.490+0900;D;SettingsActivity;onCreate(Bundle) starting
2020-11-12T23:11:54.520+0900;D;SettingsActivity;onStart() starting
2020-11-12T23:11:54.522+0900;D;SettingsActivity;onResume() starting
2020-11-12T23:11:54.909+0900;D;FileDisplayActivity;onStop() ending
2020-11-12T23:11:54.909+0900;V;FileDisplayActivity;onSaveInstanceState() start
2020-11-12T23:11:54.910+0900;D;FileDisplayActivity;onSaveInstanceState(Bundle) starting
2020-11-12T23:11:54.911+0900;D;ExtendedListFragment;onSaveInstanceState()
2020-11-12T23:11:54.911+0900;V;FileDisplayActivity;onSaveInstanceState() end
2020-11-12T23:11:56.183+0900;D;SettingsActivity;onPause() ending
2020-11-12T23:11:56.196+0900;D;LogsActivity;onCreate(Bundle) starting

````

I think it is an issue that needs to be solved. 
Because it seems like an error that does not change its name.

**Comment by tobiasKaminsky** [comment]
Is the file properly renamed when you look into it on web ui?

**Comment by aqaqsubin** [comment]
Web ui does not update automatically too.
Needs to refresh.

**Comment by willpyshan13** [comment]
> Is the file properly renamed when you look into it on web ui?

yes，when i refresh the file list ,it will be change.

**Comment by willpyshan13** [comment]
> This bug report did not receive an update in the last 4 weeks. Please take a look again and update the issue with new details, otherwise the issue will be automatically closed in 2 weeks. Thank you!

this bug isn't be resolve, why do you want to automatically closed?

**Comment by AndyScherzinger** [comment]
Thanks for the ping!

@tobiasKaminsky just rename a folder and see that the list doesn't refresh. So a pull to refresh and see the folder name being updated.

**Comment by willpyshan13** [comment]
> Thanks for the ping!
> 
> @tobiasKaminsky just rename a folder and see that the list doesn't refresh. So a pull to refresh and see the folder name being updated.

i resolve this bug use eventbus. when the rename operation is complete ,it will be invoke and refresh the list.

**Comment by willpyshan13** [comment]
i resolve this bug use eventbus. when the rename operation is complete ,it will be invoke and refresh the list.

**Comment by AndyScherzinger** [comment]
@willpyshan13 fancy opening a PR with your fix? 😊

**Comment by joshtrichards** [comment]
Fixed in #9475
